util/network/src/node_table.rs

```diff
@@ -368,6 +368,15 @@ pub fn is_valid_node_url(url: &str) -> bool {
 	Node::from_str(url).is_ok()
 }
 
+/// Same as `is_valid_node_url` but returns detailed `NetworkError`
+pub fn validate_node_url(url: &str) -> Option<NetworkError> {
```

Comment:
Why introduce another method then?
Wouldn't it be better to just do `is_valid_node_url().is_ok()` in places that don't care about the error?

Reply:
I thought about it and it was my first intention, however, in that case, method name would conflict with it's semantics. 

Also I'm a bit confused why we validate nodes by effectively doing all the work needed to construct `Node` just to throw it away and use the raw string value. Wouldn't it be better to return `Reusult<Vec<Node>, NetworkError>` from `init_reserved_nodes`? Anyway, I thought, that would cause substantial refactoring and is out of the scope.

For now I think it would be better to stick with the new function and refactor existing calls to `is_valid_node_url`.